res/controllers/Pioneer-DDJ-SX-scripts.js

```diff
@@ -0,0 +1,2149 @@
+    if (!trackLoaded) {
+        return;
+    }
+
+    if (value && !cue) {
+        // play illumination:
+        PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.playPauseDeck, true);
+        PioneerDDJSX.playFromCuePoint = true;
+        return;
+    }
+    if (!value && !cue) {
+        // pause illumination:
+        PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.playPauseDeck, false);
+        PioneerDDJSX.playFromCuePoint = false;
+        return;
+    }
+    if (value && cue) {
+        // stop and go to cue point illumination:
+        PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, true);
+        PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.playPauseDeck, true);
+        PioneerDDJSX.cuePressed[deck] = false;
+        PioneerDDJSX.playFromCuePoint = true;
+    }
+    if (!value && cue) {
+        if (PioneerDDJSX.playFromCuePoint) {
+            // play from cue point illumination:
+            PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, false);
+            PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, true);
+            PioneerDDJSX.playFromCuePoint = false;
+        } else {
+            // set cue point illumination:
+            PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, false);
+            PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, true);
+            PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, false);
+        }
+        PioneerDDJSX.cuePressed[deck] = false;
+    }
+};
+
+PioneerDDJSX.playStutterIllumination = function(value, group, control) {
+    if (value) {
+        PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, false);
+        PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, true);
+    } else {
+        PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.playPauseDeck, true);
+    }
+};
+
+PioneerDDJSX.cueLed = function(value, group, control) {
+    PioneerDDJSX.nonPadLedControl(group, PioneerDDJSX.nonPadLeds.cue, value);
+    PioneerDDJSX.nonPadLedControl(group, PioneerDDJSX.nonPadLeds.shiftCue, value);
+};
+
+PioneerDDJSX.loadLed = function(value, group, control) {
+    var deck = PioneerDDJSX.channelGroups[group];
+    PioneerDDJSX.generalLedControl(PioneerDDJSX.nonPadLeds["loadDeck" + (deck + 1)], value);
+    PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.loadedDeck, value);
+};
+
+PioneerDDJSX.reverseLed = function(value, group, control) {
+    PioneerDDJSX.nonPadLedControl(group, PioneerDDJSX.nonPadLeds.censor, value);
+    PioneerDDJSX.nonPadLedControl(group, PioneerDD
```

Comment:
You can't switch the PAD MODE in "Serato Mode" and also the pads in the default CUE PAD MODE don't work. Another thing I saw is that the vu meters act differently in "Serato Mode". I have to monitor the MIDI signals to see what happens there instead.
I will test sending MIDI signals for the jog LEDs in non-Serato mode.
I also found some code controlling the jog LEDs and sending serato-messages to the controller, I have to figure that out. It would be nice, if the LEDs act as I want them to.